### PR TITLE
add day 4 and utilities file

### PR DESCRIPTION
**Pull request by brendes6**

### CODE REVIEW

File: solutions/day4.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
+
+from utilities import input_reader
+
+xmas = "XMAS"
```
Comment (cdesjard) [inline solutions/day4.py]: Typically globally defined constants are written as all caps, i.e. this variable could be called `XMAS` or maybe better `XMAS_STR` or more generally `SEARCH_WORD` or something like that.

File: solutions/day4.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
+
+from utilities import input_reader
```
Comment (cdesjard) [inline solutions/day4.py:1]: yes

File: solutions/day4.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    elif direction in [3, 5, 8]:
+        y += 1
+    
+    return x, y
+
+
+def search(grid, x, y, word, direction):
+
+    # Essentially DFS, but instead of recursively calling for all neighbors,
+    # only call for the index in the
```
Comment (cdesjard) [inline solutions/day4.py:31]: the suspense... how is this comment supposed to end?

File: solutions/utilities.py [hunk 4]
```diff
@@ -0,0 +1,7 @@
+
+def input_reader(day: int):
+    return_list = []
+    with open(f"input_day{day}.txt", "r") as f:
```
Comment (cdesjard) [inline solutions/utilities.py:4]: This is fine, but only works if the input file is in the directory where the command is invoked.

File: solutions/day4.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
+
```
Comment (cdesjard) [inline solutions/day4.py]:
```suggestion
```

**Review by cdesjard — APPROVED** [review]
Just typographical requests

File: solutions/day4.py (hunk 3 above)
Comment (brendes6) [reply solutions/day4.py:31]: I think I was typing this in class and then my professor started talking about the final😅 I finished typing out what it's supposed to say

**Pull request merged**